### PR TITLE
Feat/add profile

### PR DESCRIPTION
**Pull request by Maianki**
### Added profile page and it's functionality
This PR includes

- [ ] Profile page
- [ ] users slice
- [ ] Edit users bio, portfolio and profile picture

### Thanks for reviewing my PR

### CODE REVIEW

File: src/features/Post/postSlice.js [hunk 1]
```diff
@@ -37,6 +38,21 @@ export const getUserAllPosts = createAsyncThunk(
   }
 );
 
+export const getAllPostsOfSingleUser = createAsyncThunk(
+  "posts/getAllPostsOfSpecificUser",
+  async ({ username }, { rejectWithValue }) => {
+    try {
+      const response = await getAllPostsOfSingleUserService(username);
+      console.log(response);
```
Comment (govindappaarun) [inline src/features/Post/postSlice.js]: suggestion: to remove logs

File: src/features/Post/postSlice.js [hunk 2]
```diff
@@ -95,47 +111,59 @@ const postSlice = createSlice({
     },
     [getUserAllPosts.fulfilled]: (state, { payload }) => {
       state.allPosts = payload;
-      state.authStatus = "success";
+      state.postStatus = "success";
```
Comment (govindappaarun) [inline src/features/Post/postSlice.js:114]: suggestion: to use constants for values instead of hardcoing

File: src/features/Post/postSlice.js [hunk 3]
```diff
@@ -95,47 +111,59 @@ const postSlice = createSlice({
...
     [deleteUserPost.pending]: (state, { payload }) => {
       state.postStatus = "loading";
     },
     [deleteUserPost.fulfilled]: (state, { payload }) => {
       state.allPosts = payload;
-      state.authStatus = "success";
+      state.postStatus = "success";
     },
     [deleteUserPost.rejected]: (state, { payload }) => {
       console.log(payload);
```
Comment (govindappaarun) [inline src/features/Post/postSlice.js]: log

**Review by govindappaarun — APPROVED** [review]
Looks Good to merge, Nice work with User Profile editing

File: src/features/Post/postSlice.js (hunk 2 above)
Comment (Maianki) [reply src/features/Post/postSlice.js:114]: Sure will make the changes.

File: src/features/Post/postSlice.js [hunk 4]
```diff
@@ -37,6 +38,21 @@ export const getUserAllPosts = createAsyncThunk(
   }
 );
 
+export const getAllPostsOfSingleUser = createAsyncThunk(
+  "posts/getAllPostsOfSpecificUser",
+  async ({ username }, { rejectWithValue }) => {
+    try {
+      const response = await getAllPostsOfSingleUserService(username);
```
Comment (vrx29) [inline src/features/Post/postSlice.js:45]: Nice that you have created services also.

**Review by vrx29 — APPROVED** [review]
Nice and clean code. Good work on folder structure, you can merge it.

**Pull request merged**